templates/core/core-cm.yaml

```diff
@@ -36,20 +43,36 @@ data:
   CLAIR_DB: "{{ template "harbor.database.clairDatabase" . }}"
   CLAIR_DB_SSLMODE: "{{ template "harbor.database.sslmode" . }}"
   CLAIR_URL: "http://{{ template "harbor.fullname" . }}-clair:6060"
+{{- if .Values.internalTLS.enabled }}
+  CLAIR_ADAPTER_URL: "https://{{ template "harbor.clair" . }}:8443"
+{{- else }}
   CLAIR_ADAPTER_URL: "http://{{ template "harbor.clair" . }}:8080"
+{{- end }}
   WITH_TRIVY: {{ .Values.trivy.enabled | quote }}
+{{- if .Values.internalTLS.enabled }}
+  TRIVY_ADAPTER_URL: "https://{{ template "harbor.trivy" . }}:8443"
+{{- else }}
   TRIVY_ADAPTER_URL: "http://{{ template "harbor.trivy" . }}:8080"
+{{- end }}
   REGISTRY_STORAGE_PROVIDER_NAME: "{{ .Values.persistence.imageChartStorage.type }}"
   WITH_CHARTMUSEUM: "{{ .Values.chartmuseum.enabled }}"
+{{- if .Values.internalTLS.enabled }}
+  CHART_REPOSITORY_URL: "https://{{ template "harbor.chartmuseum" . }}"
+{{- else }}
   CHART_REPOSITORY_URL: "http://{{ template "harbor.chartmuseum" . }}"
+{{- end }}
   LOG_LEVEL: "{{ .Values.logLevel }}"
   CONFIG_PATH: "/etc/core/app.conf"
   SYNC_REGISTRY: "false"
   CHART_CACHE_DRIVER: "redis"
   _REDIS_URL: "{{ template "harbor.redisForCore" . }}"
   _REDIS_URL_REG: "{{ template "harbor.redisForGC" . }}"
   PORTAL_URL: "http://{{ template "harbor.portal" . }}"
+{{- if .Values.internalTLS.enabled }}
```

Comment:
How about move all these checking for internal tls in the same `if else end` block?